src/core/lib/gprpp/linked_list.h

```diff
@@ -0,0 +1,149 @@
+/*
+ *
+ * Copyright 2019 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#ifndef GRPC_CORE_LIB_GPRPP_LINKED_LIST_H
+#define GRPC_CORE_LIB_GPRPP_LINKED_LIST_H
+
+#include "src/core/lib/gprpp/memory.h"
+
+namespace grpc_core {
+
+template<typename T> class LinkedList {
+ public:
+  ~LinkedList() {
+    clear();
+  }
+
+  struct ListNode {
+    template<typename... Args>
+    ListNode(Args&&... args) {
+      new (&val) T(std::forward<Args>(args)...);
+    }
+
+    ListNode *prev_ = nullptr;
+    ListNode *next_ = nullptr;
+    T val;
+  };
+
+  struct Iter
+  {
+      LinkedList<T>& list_;
+      ListNode *node_;
+
+      Iter(LinkedList<T>& list) : list_(list), node_(list.head_) {}
+      Iter(LinkedList<T>& list, ListNode *node) : list_(list), node_(node) {}
+
+      bool operator!=(const Iter& other) const
+      {
+          return !(*this == other);
+      }
+
+      bool operator==(const Iter& other) const
+      {
+          return node_ == other.node_;
+      }
+
+      Iter& operator++()
+      {
+          if (node_) {
+            node_ = node_->next_;
+          }
+          return *this;
+      }
+
+      T& operator*() const
+      {
+          return node_->val;
+      }
+  };
```

Comment:
please fix the styling of `Iter`.

Reply:
Ack; removing this class. I did add an Iter to InlinedVector that should be fine.